atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/TmManager.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.impl;
+
+import java.util.Optional;
+
+import com.palantir.atlasdb.transaction.api.TransactionManager;
+
+public interface TmManager {
```

Comment:
Same thing here.

I'm also not sure I follow the whole register vs get last registered. How often are people registering transaction managers? if you run in parallel will it ever be the case that I get the wrong transaction manager back after calling `getLastRegisteredTm`?

Reply:
I'll add some JavaDocs on usage